src/fidesops/schemas/privacy_request.py

```diff
@@ -19,6 +23,31 @@ class PrivacyRequestCreate(BaseSchema):
     requested_at: datetime
     identities: List[PrivacyRequestIdentity]
     policy_key: str
+    encryption_key: Optional[str] = None
+    nonce: Optional[str] = None
+
+    @root_validator
+    @classmethod
+    def validate_encryption_key_and_nonce(
+        cls: "PrivacyRequestCreate", values: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        """
+        Validate if encryption key or nonce is supplied, both fields must be present
+        and verify values where applicable
+        """
+        encryption_key = values.get("encryption_key")
+        nonce = values.get("nonce")
+        if nonce or encryption_key:
+            if not (nonce and encryption_key):
+                raise ValueError(
+                    "To encrypt access request data, both an encryption key and a nonce must be supplied."
+                )
+
+            # Validate supplied encryption key and nonce are acceptable
+            verify_encryption_key(encryption_key)
+            verify_nonce(nonce)
```

Comment:
I'd prefer if we don't *require* a nonce; I'd generally not expect to provide one to a system as they are meant to be random, temporary, and never re-used. It's a bit *too easy* for a caller to just hardcode a nonce and provide it with every request so I'd like to discourage that by making the default case where we generate the nonce.

That said, we will also need to change the encrypted payload to prepend the nonce (aka initialization vector, aka IV). It's super common to just do `IV || ciphertext` as the output, so we can follow that pattern - I can show an example of how this is implemented in sqlalchemy for reference...

Reply:
Sorry I missed some of the requirements discussion about this on #46!

Reply:
👍 reworking so we generate the nonce.

Reply:
@NevilleS @stevenbenjamin finished a refactor so we generate the nonce on our end, thanks for the comments -

Also added docs in the latest commit for how to decrypt.